pkg/controller/realmadmin/settings_modify.go

```diff
@@ -499,3 +505,32 @@ func parseSMSTextTemplates(r *http.Request, form *formData) {
 		}
 	}
 }
+
+// explodeSortAndDedupe explodes the given string on commas and newlines,
+// iterates over each result and removes spaces and commas, removes duplicates,
+// and returns a sorted result.
+func explodeSortAndDedupe(in string) []string {
+	pieces := strings.FieldsFunc(in, func(r rune) bool {
+		return r == '\n' || r == ','
+	})
+
+	m := make(map[string]struct{}, len(pieces))
+	for _, piece := range pieces {
+		piece := project.TrimSpaceAndNonPrintable(piece)
+		piece = strings.Trim(piece, ",")
```

Comment:
there shouldn't be a ',' in the piece since that's one of the split characters above...